com.unity.mobile.notifications/Editor/NotificationSettingsManager.cs

```diff
@@ -0,0 +1,342 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using UnityEditor;
+using UnityEngine;
+
+using Unity.Notifications.iOS;
+
+[assembly: InternalsVisibleTo("Unity.Notifications.Tests")]
+namespace Unity.Notifications
+{
+    [HelpURL("Packages/com.unity.mobile.notifications/documentation.html")]
+    internal class NotificationSettingsManager : ScriptableObject
+    {
+        internal const string ASSET_PATH = "Editor/com.unity.mobile.notifications/NotificationSettings.asset";
+
+        [SerializeField]
+        public int toolbarInt = 0;
+
+        public List<NotificationSetting> iOSNotificationSettings;
+        public List<NotificationSetting> AndroidNotificationSettings;
+
+        [SerializeField]
+        internal NotificationSettingsCollection iOSNotificationSettingsValues;
+
+        [SerializeField]
+        internal NotificationSettingsCollection AndroidNotificationSettingsValues;
+
+        private void SaveSetting(NotificationSetting setting, NotificationSettingsCollection values)
+        {
+            if (!values.Contains(setting.key) || values[setting.key].ToString() != setting.val.ToString())
+            {
+                values[setting.key] = setting.val;
+                EditorUtility.SetDirty(this);
+            }
+        }
+
+        internal void SerializeData()
+        {
+            EditorUtility.SetDirty(this);
+            AssetDatabase.SaveAssets();
```

Comment:
AssetDatabase.SaveAssets(); shouldn't be called unless really necessary, since you're saving all the assets in all Unity project.

Setting dirty is enough, Unity will then decide when it's fine to save assets